src/StreamrClient.ts

```diff
@@ -188,7 +188,7 @@ export class StreamrClient extends EventEmitter { // eslint-disable-line no-rede
     // TODO annotate connection parameter as internal parameter if possible?
     constructor(options: StreamrClientOptions = {}, connection?: StreamrConnection) {
         super()
-        this.id = counterId(`${this.constructor.name}:${uid}`)
+        this.id = counterId(`${this.constructor.name}:${uid}${options.id || ''}`)
```

Comment:
If the use provides the id, it is maybe unique enough to distinguish the instances? This `counterId(`${this.constructor.name}:${uid}`)` could be just a fallback if no explicit id is provided.

Reply:
Good point, though the wrapper around the `id` here was added to ensure I get unique results across tests, but I can give the tests known "ids":
e.g. 

```js
const publisher = new Client({id: 'publisher'})
const subscriber = new Client({id: 'subscriber'})
```
Doesn't matter so much as this is 100% for debugging/logging purposes currently, but maybe `id` isn't the right `option` property, perhaps it should be `name` or something.

Reply:
Yes, name would describe it better :thumbsup:

There is the same situation in `Subscription` class (line 53)